composite/token-allocation/src/adapter.ts

```diff
@@ -35,19 +36,34 @@ export const marketCapTotalValue = (
     .toNumber()
 }
 
-const inputParams = {
-  allocations: true,
-  quote: false,
-}
-
 const toValidAllocations = (allocations: TokenAllocations): TokenAllocations => {
   if (!allocations.every((t) => !!t.symbol))
     throw new AdapterError({ message: `Symbol not available for all tokens.`, statusCode: 400 })
-  return allocations.map((t) => ({
-    symbol: t.symbol.toUpperCase(),
-    decimals: t.decimals || DEFAULT_TOKEN_DECIMALS,
-    balance: t.balance || DEFAULT_TOKEN_BALANCE * 10 ** (t.decimals || DEFAULT_TOKEN_DECIMALS),
-  }))
+
+  const _toValidDecimals = (decimals: number) =>
+    Number.isInteger(decimals) && decimals >= 0 ? decimals : DEFAULT_TOKEN_DECIMALS
+  const _toValidBalance = (balance: BigNumberish, decimals: number) => {
+    if (!balance) {
+      return DEFAULT_TOKEN_BALANCE * 10 ** decimals
+    }
+    if (!Number(balance)) {
```

Comment:
Not using `BigNumber` because small numbers give `underflow` issues

Reply:
Could you please explain the issue here? When do you have underflows?

Also, what are you exactly checking here? When do you expect to throw `"Invalid balance"`?